core/src/main/scala/org/apache/spark/scheduler/BlacklistStrategy.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+import scala.collection.mutable
+
+import org.apache.spark.Logging
+import org.apache.spark.SparkConf
+import org.apache.spark.util.SystemClock
+
+/**
+ * The interface to determine executor blacklist and node blacklist.
+ */
+trait BlacklistStrategy {
+  val expireTimeInMillisecond: Long
+
+  // TaskId is optional here because the BlacklistStrategy could be unrelated to TaskId
```

Comment:
ah, I see now.  I was originally referring to both `getExecutorBlacklist` and `getNodeBlacklist` -- thanks for pointing out that `getNodeBlacklist` actually is used w/out a `taskId`, I had missed that.

I'd like to avoid over-engineering this api -- it'll be more clear to future readers & easier to test if we keep it minimal, so I am thinking that maybe we should just have two methods:

```scala
def getExecutorBlacklist(
    failedExecutorMap: mutable.HashMap[String, FailureStatus],
    taskId: Long): Set[String]
)

def getNodeBlacklist(failedExecutorMap: mutable.HashMap[String, FailureStatus]): Set[String]
```

that is, `getExecutorBlacklist` always takes a `taskId`, and `getNodeBlacklist` never does.  Does that support the required use case?  (Docs would also really help explain the purpose of these methods.)  Feel free to push back if I'm still missing something.